Measure distance

ram-sharan25 opened the pull request:
### Measure distance on the map.
Changes 
- Updated readme.
- added `webpack-dev-server` package to support Live Reloading.
- Added a new measure icon to the map toolbar.
- Implemented event listeners for mouse clicks (to set the anchor) and mouse movements (to draw the line and update distance).
- Real-time calculation and rendering of the distance label above the measurement line.

paridhi-parajuli commented:
It would be better if you could change the color of the text (showing the measurement), because its hard to see over the raster. May be try putting the text over a box with white background.
@ram-sharan25 @sanzog03

ram-sharan25 commented:
Fixes :

- Fixed bugs related to the eraser tool.
- Updated the measurement label's position to the tip of the line, ensuring visibility regardless of the line's length or position. This also resolved the issue where labels weren't shown for very short distances.Previously for shorter distance line length was not long enough to anchor the label on the map. 
- Changed the measurement label color from blue to white with a black border, aligning it with other labels on the map for consistency.
@paridhi-parajuli  @amarouane-ABDELHAK  @sanzog03

sanzog03 commented on emit-ch4plume-v1/src/index.js:
```diff
@@ -25,525 +25,772 @@ const markerProps = new Object();
+}
+class ClearDistancePoints {
+  onClick() {
+    distancePoints.features.splice(0, distancePoints.features.length);
```
Is this used to clear the features list? if yes we could rather do .length = 0 for a constant time operation.

sanzog03 commented on emit-ch4plume-v1/src/index.js:
```diff
@@ -25,525 +25,772 @@ const markerProps = new Object();
+      "</div>";
+    return this.container;
+  }
+  onRemove() {
```
This seems to be reused in many other control classes. could create an abstract class with this `onRemove()` reused definition. Inheriting that would remove the need for this one in  `ClearDistancePoints` control class. More DRY the better.

sanzog03 commented on emit-ch4plume-v1/src/index.js:
```diff
@@ -25,525 +25,772 @@ const markerProps = new Object();
+class ClearDistancePoints {
+  onClick() {
+    distancePoints.features.splice(0, distancePoints.features.length);
+     measureLine.features.splice(0, measureLine.features.length);
```
same as above

sanzog03 commented on emit-ch4plume-v1/src/index.js:
```diff
@@ -25,525 +25,772 @@ const markerProps = new Object();
+  onClick() {
+    distancePoints.features.splice(0, distancePoints.features.length);
+     measureLine.features.splice(0, measureLine.features.length);
+     distanceLabel.features.splice(0, distanceLabel.features.length);
```
same as above

sanzog03 commented on emit-ch4plume-v1/README.md:
```diff
@@ -1,5 +1,65 @@
+
+If you use [`nvm`](https://github.com/creationix/nvm), activate the desired Node version:
+
+`nvm install`
```
add a .nvmrc file with appropriate version. 
And add `nvm use` in the readme

sanzog03 commented:
@ram-sharan25 The ruler icon when selected and unselected, looks very similar.  Need to update that to make them clear. 
@abarciauskas-bgse Also a thought, when the scale tool is active we could disable or even hide the datetime filter and search. It might help in further eliminating the edge cases to be handled.

abarciauskas-bgse commented:
@sanzog03 did you mean to tag me in this PR?

sanzog03 commented:
sorry it was a mistake!

ram-sharan25 commented:
# New Changes

- Added `.nvmrc` to specify Node.js version.
- Updated README to include usage of `.nvmrc`.
- Refactored class definition in `MapControls` for better organization.
- Added feature to toggle map scale between kilometers and miles for both measurement mode and the map overall.
- Disabled measure mode on double-click to improve user interaction.
- Disabled search and date range pickers when measure mode is active.
- Updated measure icon to make it more prominent, with a clear visual indication when switching between modes.
@sanzog03  @amarouane-ABDELHAK

amarouane-ABDELHAK commented:
Looks good to me. Thank you

amarouane-ABDELHAK commented on emit-ch4plume-v1/README.md:
```diff
@@ -1,5 +1,69 @@
+To set the environment variables from the `.env` file, add export in front of each variable. 
+For example:
+
+`export MAP_STYLE=“xxxxxxxxxxxx”`
```
Can you replace this by running 
`node --env-file .env update_data.js`
When running this command the developer should not have to export the env vars
